### DISCUSSION TITLE
Raycast

### DISCUSSION THREAD

**Discussion by ViachaslauBusel** [bepu/bepuphysics2#237] [Q&A] [answered]
Hello.
Is calling the  Simulation.RayCast method thread-safe?

**Comment by RossNordby** [comment, marked as answer]
Multiple threads can call `RayCast` at the same time, yes.

It's not safe to modify the `Simulation` asynchronously (adding/removing stuff, timestepping) at the same time as another thread is calling `Simulation.RayCast`, though.
